src/main/java/edu/umich/ctools/sectionsUtilityTool/ToolAccessGranter.java

```diff
@@ -0,0 +1,52 @@
+package edu.umich.ctools.sectionsUtilityTool;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.client.methods.HttpGet;
+import org.json.JSONObject;
+
+
+public class ToolAccessGranter {
+	private static Log m_log = LogFactory.getLog(ToolAccessGranter.class);
+	private final String courseId;
+	private final String canvasUserId;
+
+	public ToolAccessGranter(String courseId, String canvasUserId){
+		this.courseId=courseId;
+		this.canvasUserId=canvasUserId;
+	}
+
+
+	public boolean isAllowedAccess(boolean isAdmin) {
+
+		// access is given to the admin's in course account hierarchy and super admins.
+		if(isAdmin){
+			return true;
+		}
+		return checkIfUserHasPermissionToManageStudentsInCourse();
+
+
+	}
+	/*
+	 This call is checking if the user role in the course like instructor, student, TA or any other has manage student access
+	 if he does then the user is granted access to the tool.
+	 */
+
+	private boolean checkIfUserHasPermissionToManageStudentsInCourse() {
+//		https://umich.test.instructure.com/api/v1/courses/3/permissions?as_user_id=2
+		boolean canManageStudents=false;
+		String url = Utils.urlConstructor(Utils.URL_CHUNK_COURSES, courseId,"/permissions?as_user_id=",canvasUserId);
+		ApiResultWrapper arw = Utils.makeApiCall(new HttpGet(url));
+		String errMsg=String.format("The Permission check to grant access to the tool failed with status code %s due to %s in the course %s for the user %s"
```

Comment:
Suggest pulling out the message text into a separate variable so that the format call can be a lot shorter and easier to understand.

Reply:
Done